libs/metrics-core/src/Data/Metrics.hs

```diff
@@ -2,214 +2,281 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 
 module Data.Metrics
-    ( Path
+    (
+    -- * Types
+      Path
     , Metrics
+    , Histogram
     , Counter
     , Gauge
-    , Label
-    , Buckets
-
-    , path
-    , metrics
 
+    -- * Counters
     , counterGet
-    , counterGet'
     , counterAdd
     , counterIncr
     , counterValue
 
+    -- * Gauges
     , gaugeGet
-    , gaugeGet'
     , gaugeAdd
     , gaugeSub
     , gaugeIncr
     , gaugeDecr
     , gaugeSet
     , gaugeValue
 
-    , labelGet
-    , labelGet'
-    , labelSet
-    , labelValue
+    -- * Histograms
+    -- ** Types
+    , HistogramInfo
+    , Buckets
+    , Bucket
+
+    -- ** Describing Histograms
+    , linearHistogram
+    , customHistogram
+    , deprecatedRequestDurationHistogram
 
-    , bucketsGet
-    , bucketsGet'
-    , bucketsIncr
+    -- ** Manipulating Histograms
+    , histoGet
+    , histoSubmit
+    , histoValue
+    , histoTimeAction
 
+    -- * Helper functions
+    , path
+    , metrics
     , render
     ) where
 
 import Imports hiding (lookup, union)
 import Data.Aeson
-import Data.Atomics.Counter (AtomicCounter)
 import Data.Hashable
-import Data.Metrics.Buckets (Buckets)
 
-import qualified Data.Atomics.Counter as Atomic
 import qualified Data.Text            as T
-import qualified Data.HashMap.Strict  as Map
-import qualified Data.Metrics.Buckets as Buckets
+import qualified Data.HashMap.Strict  as HM
+import qualified Data.Map.Strict      as M
 import qualified Data.Metrics.GC      as GC
-
-newtype Path = Path { _path :: Text } deriving (Eq, Hashable, Semigroup, Monoid)
-
+import qualified Prometheus as P
+
+-- | Internal Counter type
```

Comment:
you could have just imported those as qualified.  but perhaps your way is clearer?

Reply:
I struggle to find the right line here; I want to express that users shouldn't be using the Prometheus types directly; i.e. they should only use the interface which we expose here; but I suppose that without newtypes that's not enforced. Maybe I should just switch to use newtypes :+1: